python_modules/dagster/dagster/_core/definitions/external_asset.py

```diff
@@ -181,3 +181,12 @@ def _shim_assets_def(context: AssetExecutionContext):
             return return_value
 
     return _shim_assets_def
+
+
+def create_unexecutable_external_assets_from_assets_def(
+    assets_def: AssetsDefinition,
+) -> Sequence[AssetsDefinition]:
+    if not assets_def.is_executable:
+        return [assets_def]
+    else:
+        return [create_external_asset_from_source_asset(sa) for sa in assets_def.to_source_assets()]
```

Comment:
I'm struggling to connect this name to the behavior.

When can an `assets_def` returns `True` out of `is_executable` contain source assets?

Reply:
All assets defs can be _converted_ to source assets-- none of them "contain" source assets. This behavior has been used to make assets available for loading to jobs that don't materialize them. We're just doing the same thing here but using an unexecutable assets def for the loadable representation instead of a source asset.

This is important because the assets defs are the source of truth for what assets are being materialized by a job. Previously it was roughly "if there is an assets def for it it's being materialized". That logic no longer applies because loadable assets are now also represented by assets defs-- so it has to be "if there is a _materializable_ assets def for it it's being materialized". So when we want to make an existing materialiazble assets def available for loading, we convert it to unexecutable.

Reply:
That all makes sense.

When the dust settles from all this you should write a document describing the new ontology. Will be a super useful reference.

`create_unexecutable_representation_of_assets_def` or a name like could be more clear?

Reply:
It's kind of a tough one to name, but I prefer the current name because:

- it's consistent with the other `create_external_asset...` names in this module
- "unexecutable representation of assets def" sounds kind of like it could be creating something that isn't an assets def. Also it would maybe need to be "representations" since we are returning multiple assets def

Reply:
Yeah nothing is totally satisfying. This name plus an explanatory comment suffices.

Reply:
added comment